src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
```

Comment:
PullProofAI

The `getUserData` method should use parameterized queries to prevent SQL injection vulnerabilities.
```diff
+ const query = `SELECT * FROM users WHERE id = ?`;
+ return this.executeQuery(query, [userId]);
```



<!-- This is an auto-generated comment by PullProofAI -->